bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -1295,4 +1298,127 @@ public boolean equals(Object otherPair) {
             return true;
         }
     }
+
+    /**
+     * Checks whether a given {@link Byte} has an implicit initial value or not.
+     * @param type {@link Byte} to be analyzed.
+     * @return whether there's an implicit initial value or not.
+     */
+    public static boolean hasImplicitInitialValue(BType type) {
+        return hasImplicitInitialValue(type, new ArrayList<>());
+    }
+
+    private static boolean hasImplicitInitialValue(BType type, List<BType> unAnalyzedTypes) {
+        if (type == null) {
+            return true;
+        }
+        if (type.getTag() < TypeTags.RECORD_TYPE_TAG) {
+            return true;
+        }
+        switch (type.getTag()) {
+            case TypeTags.TYPEDESC_TAG:
+            case TypeTags.STREAM_TAG:
+            case TypeTags.MAP_TAG:
+            case TypeTags.ANY_TAG:
+                return true;
+            case TypeTags.ARRAY_TAG:
+                return hasImplicitInitialValue(((BArrayType) type).getElementType());
+            case TypeTags.FINITE_TYPE_TAG:
+                return analyzeFiniteType((BFiniteType) type);
+            case TypeTags.OBJECT_TYPE_TAG:
+                return analyzeObjectType((BObjectType) type);
+            case TypeTags.RECORD_TYPE_TAG:
+                return analyzeRecordType((BRecordType) type, unAnalyzedTypes);
+            case TypeTags.TUPLE_TAG:
+                BTupleType tupleType = (BTupleType) type;
+                return tupleType.getTupleTypes().stream().allMatch(TypeChecker::hasImplicitInitialValue);
+            case TypeTags.UNION_TAG:
+                return analyzeUnionType((BUnionType) type);
+            default:
+                return false;
+        }
+    }
+
+    private static boolean analyzeUnionType(BUnionType type) {
```

Comment:
Shall we rename this to indicate we are checking for the existence of a filler value?